src/EditorFeatures/Test/Tagging/AsynchronousTaggerTests.cs

```diff
@@ -94,6 +96,8 @@ void M()
         {
             using (var workspace = await CSharpWorkspaceFactory.CreateWorkspaceFromFileAsync("class Program {\r\n\r\n}"))
             {
+                WpfTestCase.RequireWpfFact($"{nameof(AsynchronousTaggerTests)}.{nameof(TestSynchronousOutlining)} creates asynchronous taggers");
```

Comment:
Foreground/background threading logic. (Consider the code that schedules TagsChanged.)